src/transformers/models/clvp/modeling_clvp.py

```diff
@@ -0,0 +1,1873 @@
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+            output_hidden_states (`bool`, *optional*):
+                Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors
+                for more detail.
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+
+        Examples:
+
+        ```python
+        >>> from transformers import CLVPTransformerWithProjection, CLVPTextConfig, CLVPTokenizer
+
+        >>> text_config = CLVPTextConfig.from_pretrained("susnato/clvp_dev")
+        >>> model = CLVPTransformerWithProjection(text_config)
+        >>> tokenizer = CLVPTokenizer.from_pretrained("susnato/clvp_dev")
+
+        >>> text = "This is an example text."
+        >>> inputs = tokenizer(text, return_tensors="pt")
+
+        >>> outputs = model(**inputs)
+        ```
+        """
+
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        transformer_outputs = self.transformer(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            use_causal_attention_mask=use_causal_attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        pooled_output = transformer_outputs[1]
+
+        embeds = self.projection(pooled_output)
+
+        if not return_dict:
+            outputs = (embeds,) + transformer_outputs
+            return outputs
+
+        return CLVPTransformerWithProjectionOutput(
+            embeds=embeds,
+            last_hidden_state=transformer_outputs.last_hidden_state,
+            pooler_output=transformer_outputs.pooler_output,
+            hidden_states=transformer_outputs.hidden_states,
+            attentions=transformer_outputs.attentions,
+        )
+
+
+# Copied from transformers.models.gpt2.modeling_gpt2.GPT2MLP with GPT2->CLVPAutoRegressive
+class CLVPAutoRegressiveMLP(nn.Module):
+    def __init__(self, intermediate_size, config):
+        super().__init__()
+        embed_dim = config.hidden_size
+        self.c_fc = Conv1D(intermediate_size, embed_dim)
+        self.c_proj = Conv1D(embed_dim, intermediate_size)
+        self.act = ACT2FN[config.activation_function]
+        self.dropout = nn.Dropout(config.resid_pdrop)
+
+    def forward(self, hidden_states: Optional[Tuple[torch.FloatTensor]]) -> torch.FloatTensor:
+        hidden_states = self.c_fc(hidden_states)
+        hidden_states = self.act(hidden_states)
+        hidden_states = self.c_proj(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        return hidden_states
+
+
+class CLVPAutoRegressiveBlock(nn.Module):
```

Comment:
This is the decoder no?

Reply:
yes this is the decoder.

Reply:
no, this is almost similar to the [GPT2Block](https://github.com/huggingface/transformers/blob/866df66fe4a3b3b948b926d849ef291675b8a24c/src/transformers/models/gpt2/modeling_gpt2.py#L361)